modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -3785,7 +3785,7 @@ private void process(MouseEvent e, boolean onPulse) {
                             backButtonDown || forwardButtonDown)) {
                         //old gesture ended and new one started
                         gestureStarted = true;
-                        if (!PLATFORM_DRAG_GESTURE_INITIATION) {
+                        if (!PLATFORM_DRAG_GESTURE_INITIATION && Scene.this.dndGesture == null) {
```

Comment:
This will prevent the creation of the new `DndGesture`, but we will still clear the PDR targets. Have you confirmed that this is the desired behavior? It seems reasonable, since we also clear it on a mouse move, but wanted to get your take on it.

Reply:
This is really an edge case. There can be two gestures at once, an existing one (i.e from TouchPad), with its full drag information, that hasn't finished yet, and a new mouse one. 

Having a single `dndGesture` object for both events seems wrong. But having a list/array of events for such scenario seems overkill. 

Somehow some information from one or the other event might get lost or mixed up. The PDR targets are not required for the initial touch event, and we prevent the RTE, so I'd say this is a reasonable non-intrusive fix.

Reply:
In looking at it more closely, I think it might be better to avoid setting `gestureStarted` if there is already a drag gesture in process (started via dragEnter). Maybe something like this?

```
    if (Scene.this.dndGesture == null) {
        //old gesture ended and new one started
        gestureStarted = true;
        if (!PLATFORM_DRAG_GESTURE_INITIATION) {
            Scene.this.dndGesture = new DnDGesture();
        }
        clearPDREventTargets();
    }
```

Reply:
Makes sense, I'll change it, but avoiding two nested if expressions.